solr/core/src/java/org/apache/solr/query/FilterQuery.java

```diff
@@ -89,13 +87,13 @@ public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float bo
 
     if (!(searcher instanceof SolrIndexSearcher)) {
       // delete-by-query won't have SolrIndexSearcher
-      return new BoostQuery(new ConstantScoreQuery(q), 0).createWeight(searcher, scoreMode, 1f);
+      return new ConstantScoreQuery(q).createWeight(searcher, scoreMode, 1f);
```

Comment:
@dsmiley Does this take care of SOLR-14800 as well?

Reply:
ConstantScoreWeight is abstract, we'd have to implement several more methods if we did that. Wouldn't be as concise I think.

Reply:
Good point.